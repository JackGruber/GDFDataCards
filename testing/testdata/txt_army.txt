++ 300pt - PB - Prime Brothers [GFF 300pts] ++

Prime Master test [1] Q3+ D2+ | 75pts | Fearless, Hero, Tough(3)
CCW (A2), Heavy Rifle (24", A1, AP(1))

Infiltrator [1] Q4+ D3+ | 40pts | Fearless, Scout, Strider, Scopes(Sniper)
CCW (A2), Marksman Carbine (18", A1, Rending)

Assault Prime Brother [1] Q3+ D2+ | 40pts | Fearless
CCW (A3), Plasma Pistol (12", A1, AP(4))

Prime Brother [1] Q3+ D2+ | 50pts | Fearless
Auto-Rifle (18", A2), CCW (A2), Grenade-Mod (12", A1, Blast(3))

Prime Brother [1] Q3+ D2+ | 35pts | Fearless
CCW (A2), Heavy Rifle (24", A1, AP(1))

Eliminator Prime Brother with more text [1] Q3+ D3+ | 60pts | Fearless, Scout, Stealth
CCW (A2), Sniper Rifle (30", A1, AP(1), Sniper)

Storm Leader [1] Q4+ D4+ | 85pts | Hero, Tough(3), Strider, 1x Forward Observer(Take Aim)
CCW (A2), Plasma Pistol (12", A1, AP(4))

Veteran [1] Q4+ D5+ | 20pts | 1x Demo Charges(AP(4) in melee)
CCW (A1), Rifle (24", A1)

2x Veteran [1] Q4+ D5+ | 15pts | 
CCW (A1), Rifle (24", A1)

Special Weapon [1] Q5+ D5+ | 25pts | Relentless
CCW (A1), Flamethrower (12", A6)

Special Weapon [1] Q5+ D5+ | 25pts | Relentless
CCW (A1), Grenade Launcher (24", A1, Blast(3))

Storm Trooper [1] Q4+ D4+ | 45pts | Strider
Energy Fist (A2, AP(4)), Shred Rifle (18", A3, Rending)

Storm Trooper [1] Q4+ D4+ | 25pts | Strider
CCW (A1), Plasma Rifle (24", A1, AP(4))

Sniper [1] Q5+ D5+ | 45pts | Scout, Stealth
CCW (A1), Sniper Rifle (30", A1, AP(1), Sniper)

Aggro Squad [3] Q3+ D2+ | 320pts | Fearless, Tough(3)
3x Dual Energy Fists (A4, AP(4)), 3x Twin Fist-Pistols (12", A4)

Prime ATV [1] Q3+ D2+ | 255pts | Fast, Fearless, Scout, Strider, Tough(6)
Heavy Fusion Rifle (18", A1, AP(4), Deadly(6)), Twin Auto-Rifles (18", A4)

Attack Speeder [1] Q3+ D2+ | 385pts | Ambush, Fast, Fearless, Impact(6), Strider, Tough(9)
Fusion Destroyer (12", A2, AP(4), Deadly(3)), Twin Strike Missiles (18", A2, AP(3), Deadly(3))

Thaddeus Taho [1] Q4+ D4+ | 75pts | Fearless, Hero, Tough(3), 1x Feudal Steed(Fast, Impact(1)), 1x Strategist(Ballistics Console)
Energy Sword (A2, AP(1), Rending), Rifle (24", A1)
