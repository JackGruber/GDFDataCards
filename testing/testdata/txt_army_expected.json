{
    "parser": "txt",
    "listName": "300pt - PB - Prime Brothers",
    "gameSystem": "gff",
    "gameSystemId": 3,
    "listPoints": 300,
    "units": [
        {
            "cost": "75",
            "specialRules": [
                {
                    "label": "Fearless",
                    "key": "fearless"
                },
                {
                    "label": "Hero",
                    "key": "hero"
                },
                {
                    "label": "Tough(3)",
                    "key": "tough",
                    "rating": "3"
                }
            ],
            "equipment": [],
            "id": "2",
            "name": "Prime Master test",
            "size": 1,
            "quality": 3,
            "defense": 2,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 2
                },
                {
                    "name": "Heavy Rifle",
                    "count": 1,
                    "range": 24,
                    "attacks": 1,
                    "ap": 1
                }
            ]
        },
        {
            "cost": "40",
            "specialRules": [
                {
                    "label": "Fearless",
                    "key": "fearless"
                },
                {
                    "label": "Scout",
                    "key": "scout"
                },
                {
                    "label": "Strider",
                    "key": "strider"
                }
            ],
            "equipment": [
                {
                    "name": "Scopes",
                    "specialRules": [
                        {
                            "label": "Sniper",
                            "key": "sniper"
                        }
                    ]
                }
            ],
            "id": "5",
            "name": "Infiltrator",
            "size": 1,
            "quality": 4,
            "defense": 3,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 2
                },
                {
                    "name": "Marksman Carbine",
                    "count": 1,
                    "range": 18,
                    "attacks": 1,
                    "specialRules": [
                        {
                            "label": "Rending",
                            "key": "rending"
                        }
                    ]
                }
            ]
        },
        {
            "cost": "40",
            "specialRules": [
                {
                    "label": "Fearless",
                    "key": "fearless"
                }
            ],
            "equipment": [],
            "id": "8",
            "name": "Assault Prime Brother",
            "size": 1,
            "quality": 3,
            "defense": 2,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 3
                },
                {
                    "name": "Plasma Pistol",
                    "count": 1,
                    "range": 12,
                    "attacks": 1,
                    "ap": 4
                }
            ]
        },
        {
            "cost": "50",
            "specialRules": [
                {
                    "label": "Fearless",
                    "key": "fearless"
                }
            ],
            "equipment": [],
            "id": "11",
            "name": "Prime Brother",
            "size": 1,
            "quality": 3,
            "defense": 2,
            "weapons": [
                {
                    "name": "Auto-Rifle",
                    "count": 1,
                    "range": 18,
                    "attacks": 2
                },
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 2
                },
                {
                    "name": "Grenade-Mod",
                    "count": 1,
                    "range": 12,
                    "attacks": 1,
                    "specialRules": [
                        {
                            "label": "Blast(3)",
                            "key": "blast",
                            "rating": "3"
                        }
                    ]
                }
            ]
        },
        {
            "cost": "35",
            "specialRules": [
                {
                    "label": "Fearless",
                    "key": "fearless"
                }
            ],
            "equipment": [],
            "id": "14",
            "name": "Prime Brother",
            "size": 1,
            "quality": 3,
            "defense": 2,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 2
                },
                {
                    "name": "Heavy Rifle",
                    "count": 1,
                    "range": 24,
                    "attacks": 1,
                    "ap": 1
                }
            ]
        },
        {
            "cost": "60",
            "specialRules": [
                {
                    "label": "Fearless",
                    "key": "fearless"
                },
                {
                    "label": "Scout",
                    "key": "scout"
                },
                {
                    "label": "Stealth",
                    "key": "stealth"
                }
            ],
            "equipment": [],
            "id": "17",
            "name": "Eliminator Prime Brother with more text",
            "size": 1,
            "quality": 3,
            "defense": 3,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 2
                },
                {
                    "name": "Sniper Rifle",
                    "count": 1,
                    "range": 30,
                    "attacks": 1,
                    "ap": 1,
                    "specialRules": [
                        {
                            "label": "Sniper",
                            "key": "sniper"
                        }
                    ]
                }
            ]
        },
        {
            "cost": "85",
            "specialRules": [
                {
                    "label": "Hero",
                    "key": "hero"
                },
                {
                    "label": "Tough(3)",
                    "key": "tough",
                    "rating": "3"
                },
                {
                    "label": "Strider",
                    "key": "strider"
                }
            ],
            "equipment": [
                {
                    "name": "Forward Observer",
                    "specialRules": [
                        {
                            "label": "Take Aim",
                            "key": "take aim"
                        }
                    ]
                }
            ],
            "id": "20",
            "name": "Storm Leader",
            "size": 1,
            "quality": 4,
            "defense": 4,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 2
                },
                {
                    "name": "Plasma Pistol",
                    "count": 1,
                    "range": 12,
                    "attacks": 1,
                    "ap": 4
                }
            ]
        },
        {
            "cost": "20",
            "specialRules": [],
            "equipment": [
                {
                    "name": "Demo Charges",
                    "specialRules": [
                        {
                            "label": "AP(4) in melee",
                            "key": "ap",
                            "rating": "4"
                        }
                    ]
                }
            ],
            "id": "23",
            "name": "Veteran",
            "size": 1,
            "quality": 4,
            "defense": 5,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 1
                },
                {
                    "name": "Rifle",
                    "count": 1,
                    "range": 24,
                    "attacks": 1
                }
            ]
        },
        {
            "cost": "15",
            "specialRules": [],
            "equipment": [],
            "id": "26",
            "name": "2x Veteran",
            "size": 1,
            "quality": 4,
            "defense": 5,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 1
                },
                {
                    "name": "Rifle",
                    "count": 1,
                    "range": 24,
                    "attacks": 1
                }
            ]
        },
        {
            "cost": "25",
            "specialRules": [
                {
                    "label": "Relentless",
                    "key": "relentless"
                }
            ],
            "equipment": [],
            "id": "29",
            "name": "Special Weapon",
            "size": 1,
            "quality": 5,
            "defense": 5,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 1
                },
                {
                    "name": "Flamethrower",
                    "count": 1,
                    "range": 12,
                    "attacks": 6
                }
            ]
        },
        {
            "cost": "25",
            "specialRules": [
                {
                    "label": "Relentless",
                    "key": "relentless"
                }
            ],
            "equipment": [],
            "id": "32",
            "name": "Special Weapon",
            "size": 1,
            "quality": 5,
            "defense": 5,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 1
                },
                {
                    "name": "Grenade Launcher",
                    "count": 1,
                    "range": 24,
                    "attacks": 1,
                    "specialRules": [
                        {
                            "label": "Blast(3)",
                            "key": "blast",
                            "rating": "3"
                        }
                    ]
                }
            ]
        },
        {
            "cost": "45",
            "specialRules": [
                {
                    "label": "Strider",
                    "key": "strider"
                }
            ],
            "equipment": [],
            "id": "35",
            "name": "Storm Trooper",
            "size": 1,
            "quality": 4,
            "defense": 4,
            "weapons": [
                {
                    "name": "Energy Fist",
                    "count": 1,
                    "attacks": 2,
                    "ap": 4
                },
                {
                    "name": "Shred Rifle",
                    "count": 1,
                    "range": 18,
                    "attacks": 3,
                    "specialRules": [
                        {
                            "label": "Rending",
                            "key": "rending"
                        }
                    ]
                }
            ]
        },
        {
            "cost": "25",
            "specialRules": [
                {
                    "label": "Strider",
                    "key": "strider"
                }
            ],
            "equipment": [],
            "id": "38",
            "name": "Storm Trooper",
            "size": 1,
            "quality": 4,
            "defense": 4,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 1
                },
                {
                    "name": "Plasma Rifle",
                    "count": 1,
                    "range": 24,
                    "attacks": 1,
                    "ap": 4
                }
            ]
        },
        {
            "cost": "45",
            "specialRules": [
                {
                    "label": "Scout",
                    "key": "scout"
                },
                {
                    "label": "Stealth",
                    "key": "stealth"
                }
            ],
            "equipment": [],
            "id": "41",
            "name": "Sniper",
            "size": 1,
            "quality": 5,
            "defense": 5,
            "weapons": [
                {
                    "name": "CCW",
                    "count": 1,
                    "attacks": 1
                },
                {
                    "name": "Sniper Rifle",
                    "count": 1,
                    "range": 30,
                    "attacks": 1,
                    "ap": 1,
                    "specialRules": [
                        {
                            "label": "Sniper",
                            "key": "sniper"
                        }
                    ]
                }
            ]
        },
        {
            "cost": "320",
            "specialRules": [
                {
                    "label": "Fearless",
                    "key": "fearless"
                },
                {
                    "label": "Tough(3)",
                    "key": "tough",
                    "rating": "3"
                }
            ],
            "equipment": [],
            "id": "44",
            "name": "Aggro Squad",
            "size": 3,
            "quality": 3,
            "defense": 2,
            "weapons": [
                {
                    "name": "Dual Energy Fists",
                    "count": 3,
                    "attacks": 4,
                    "ap": 4
                },
                {
                    "name": "Twin Fist-Pistols",
                    "count": 3,
                    "range": 12,
                    "attacks": 4
                }
            ]
        },
        {
            "cost": "255",
            "specialRules": [
                {
                    "label": "Fast",
                    "key": "fast"
                },
                {
                    "label": "Fearless",
                    "key": "fearless"
                },
                {
                    "label": "Scout",
                    "key": "scout"
                },
                {
                    "label": "Strider",
                    "key": "strider"
                },
                {
                    "label": "Tough(6)",
                    "key": "tough",
                    "rating": "6"
                }
            ],
            "equipment": [],
            "id": "47",
            "name": "Prime ATV",
            "size": 1,
            "quality": 3,
            "defense": 2,
            "weapons": [
                {
                    "name": "Heavy Fusion Rifle",
                    "count": 1,
                    "range": 18,
                    "attacks": 1,
                    "ap": 4,
                    "specialRules": [
                        {
                            "label": "Deadly(6)",
                            "key": "deadly",
                            "rating": "6"
                        }
                    ]
                },
                {
                    "name": "Twin Auto-Rifles",
                    "count": 1,
                    "range": 18,
                    "attacks": 4
                }
            ]
        },
        {
            "cost": "385",
            "specialRules": [
                {
                    "label": "Ambush",
                    "key": "ambush"
                },
                {
                    "label": "Fast",
                    "key": "fast"
                },
                {
                    "label": "Fearless",
                    "key": "fearless"
                },
                {
                    "label": "Impact(6)",
                    "key": "impact",
                    "rating": "6"
                },
                {
                    "label": "Strider",
                    "key": "strider"
                },
                {
                    "label": "Tough(9)",
                    "key": "tough",
                    "rating": "9"
                }
            ],
            "equipment": [],
            "id": "50",
            "name": "Attack Speeder",
            "size": 1,
            "quality": 3,
            "defense": 2,
            "weapons": [
                {
                    "name": "Fusion Destroyer",
                    "count": 1,
                    "range": 12,
                    "attacks": 2,
                    "ap": 4,
                    "specialRules": [
                        {
                            "label": "Deadly(3)",
                            "key": "deadly",
                            "rating": "3"
                        }
                    ]
                },
                {
                    "name": "Twin Strike Missiles",
                    "count": 1,
                    "range": 18,
                    "attacks": 2,
                    "ap": 3,
                    "specialRules": [
                        {
                            "label": "Deadly(3)",
                            "key": "deadly",
                            "rating": "3"
                        }
                    ]
                }
            ]
        },
        {
            "cost": "75",
            "specialRules": [
                {
                    "label": "Fearless",
                    "key": "fearless"
                },
                {
                    "label": "Hero",
                    "key": "hero"
                },
                {
                    "label": "Tough(3)",
                    "key": "tough",
                    "rating": "3"
                }
            ],
            "equipment": [
                {
                    "name": "Feudal Steed",
                    "specialRules": [
                        {
                            "label": "Fast",
                            "key": "fast"
                        },
                        {
                            "label": " Impact(1)",
                            "key": " impact",
                            "rating": "1"
                        }
                    ]
                },
                {
                    "name": "Strategist",
                    "specialRules": [
                        {
                            "label": "Ballistics Console",
                            "key": "ballistics console"
                        }
                    ]
                }
            ],
            "id": "53",
            "name": "Thaddeus Taho",
            "size": 1,
            "quality": 4,
            "defense": 4,
            "weapons": [
                {
                    "name": "Energy Sword",
                    "count": 1,
                    "attacks": 2,
                    "ap": 1,
                    "specialRules": [
                        {
                            "label": "Rending",
                            "key": "rending"
                        }
                    ]
                },
                {
                    "name": "Rifle",
                    "count": 1,
                    "range": 24,
                    "attacks": 1
                }
            ]
        }
    ]
}